### PR TITLE
[SW-485] transactions labels

### PR DESCRIPTION
**Pull request by ifaouibadi**

### CODE REVIEW

File: src/composables/transactionToken.ts
Comment (peronczyk) [inline src/composables/transactionToken.ts:108]: 1. I suggest to type the returned value as there are multiple returns and there is no easy way to establish this.

2. Unnecessary comment start at the beginning.
3. Rather subjective, but maybe having longer name would be meaningful for this computed property?

File: src/popup/components/TransactionItem.vue
Comment (peronczyk) [inline src/popup/components/TransactionItem.vue:90]: I guess this can be removed.

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:109]: :)

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:107]: I suggest to avoid using magic strings.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/composables/transactionLabels.ts
Comment (peronczyk) [inline src/composables/transactionLabels.ts:51]: I suggest to avoid using magic strings.

File: src/composables/transactionLabels.ts
Comment (peronczyk) [inline src/composables/transactionLabels.ts:87]: I suggest to avoid using magic strings.

File: src/composables/transactionLabels.ts
Comment (peronczyk) [inline src/composables/transactionLabels.ts:20]: I suggest to type this with the `IAccount` interface.

File: src/composables/transactionLabels.ts
Comment (peronczyk) [inline src/composables/transactionLabels.ts:19]: I suggest to type this with `INetwork` interface.

File: src/popup/components/TransactionInfo.vue
Comment (peronczyk) [inline src/popup/components/TransactionInfo.vue:79]:
```suggestion
    tx: { type: Object as PropType<ITransaction>, default: null },
```

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:41]: I suggest to avoid using `any` to type things that can be typed. Also if this computed is not an `IAccount` I suggest to rename it to something like `parsedAccount` or `computedAccount` or `accountFor...`.

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:39]: As this is not an actual function I suggest to rename it to `prefered...` (remove "get").

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/composables/transactionLabels.ts
Comment (konradodo) [inline src/composables/transactionLabels.ts:28]: After RxJs @ifaouibadi  you cant use useGetter and useState in composable

File: src/composables/transactionLabels.ts
Comment (konradodo) [inline src/composables/transactionLabels.ts:25]:
```suggestion
  transaction?: ITransaction,
```

File: src/composables/transactionLabels.ts
Comment (konradodo) [inline src/composables/transactionLabels.ts:50]:
```suggestion
    if (!(transaction && transaction.tx)) {
     return defaultLabels;
     }
```

File: src/popup/components/TransactionInfo.vue
Comment (pstachel) [inline src/popup/components/TransactionInfo.vue:87]:
```suggestion
      { tx: props.tx } as ITx,
```

File: src/popup/components/TransactionOverview.vue
Comment (pstachel) [inline src/popup/components/TransactionOverview.vue:44]:
```suggestion
    const getPreferredName = useGetter('names/getPreferred');
```

File: src/popup/components/TransactionOverview.vue
Comment (pstachel) [inline src/popup/components/TransactionOverview.vue:117]: you could use `const middleware = useState('middleware')`

**Review by pstachel — CHANGES_REQUESTED** [review]

**Review by pstachel — APPROVED** [review]

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/TransactionItem.vue
Comment (peronczyk) [inline src/popup/components/TransactionItem.vue:52]: Is `toUpperCase()` still required in line 45?

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:35]: After switching to TypeScript I suggest to use `ITx` PropType.

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:47]: For the computed properties I suggest to use the type for the return. So in this case:
```
computed((): IAccount => ...
```
But in this case you won't be able to use the `IAccount` because of missing `label` and `url`. So new (eg.: IAccountExtended`) type should be created.

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:59]: I suggest to add proper type to the function return.

File: src/popup/utils/constants.ts
Comment (peronczyk) [inline src/popup/utils/constants.ts:456]: It's really nice that we have TypeScript right now, because some inconsistencies can be noticed. Like for example the possible values for the `transaction.tx.function`. Unitil now it should use the `TxFunction` type located in `src/types/index.ts`.
But here you listed other options, in which some of them are the same.
Please establish the correct values for the `transaction.tx.function`, update the `TxFunction` type and use it to create the list of constants. Or (which I suggest) do the opposite:

```
export const TX_FUNCTION = {
  tip: 'tip',
  retip: 'retip',
  ...
} as const;
```

and then in the types file:
```
import { TX_FUNCTION } ...

export type TxFunction = ObjectValues<typeof TX_FUNCTION>;
```

File: src/popup/components/TransactionOverview.vue
Comment (peronczyk) [inline src/popup/components/TransactionOverview.vue:119]: I suggest to use `validateHash` util function:
```
const { valid, prefix } = validateHash(props.tx.recipientId);
if (valid && prefix === HASH_PREFIX_NAME) {
  ...
```

(prefered) Or create separate util function similar to `isContract` that would wrap above logic.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/TransactionItem.vue
Comment (ifaouibadi) [reply src/popup/components/TransactionItem.vue:52]: replaced with CSS style

**Comment by subhod-i** [comment]
@ifaouibadi the transaction labels are handled differently in the new SDK version. It's now an Enum of its own. We should probably wait for the SDK PR to get merged before merging this

File: src/popup/components/TransactionInfoDetailsParty.vue
Comment (pstachel) [inline src/popup/components/TransactionInfoDetailsParty.vue:13]: It can't be resolved like that - the design will break. Check out the solution from SW-493.
https://github.com/aeternity/superhero-wallet/pull/1763/files#diff-81e706951f062c4bd205ee4449b419270ae942e6161be2d6936446a537c0da64R14

File: src/composables/transactionToken.ts
Comment (peronczyk) [inline src/composables/transactionToken.ts:127]: If the type `TranslateResult[]` is put in square brackets after the `computed` it means that we are just guessing what will be returned. So this code for example will be correct:
```
const a = computed<number>(() => 'lorem ipsum');
```

In order to properly type this I suggest to type the returned function. This way the returned body will also by validated:
```
computed((): (string | TranslateResult)[] => ...
```

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/types/index.ts
Comment (peronczyk) [inline src/types/index.ts:24]: This is not that simple. In some places we are using `snake_case` names and in other `camelCase`. I had a problem with this in one of my incoming PRs.

I solved this by using this types:
```
/**
 * TxFunction names coming directly from the API or ready to be sent.
 */
export type TxFunctionRaw = ObjectValues<typeof TX_FUNCTIONS>;

/**
 * TxFunctions used internally by the app.
 */
export type TxFunctionParsed = keyof typeof TX_FUNCTIONS;

export type TxFunction = TxFunctionRaw | TxFunctionParsed;
```

Here's the change I've made:
https://github.com/aeternity/superhero-wallet/pull/1780/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Review by peronczyk — APPROVED** [review]

**Pull request merged**
